lib/project_types/script/layers/infrastructure/languages/javascript_task_runner.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+module Script
+  module Layers
+    module Infrastructure
+      module Languages
+        class JavaScriptTaskRunner
+          BYTECODE_FILE = "build/%{name}.wasm"
+          METADATA_FILE = "build/metadata.json"
+          SCRIPT_SDK_BUILD = "npm run build"
+          GEN_METADATA = "npm run gen-metadata"
+
+          attr_reader :ctx, :script_name
+
+          def initialize(ctx, script_name)
+            @ctx = ctx
+            @script_name = script_name
+          end
+
+          def build
+            compile
+            bytecode
+          end
+
+          def compiled_type
+            "wasm"
+          end
+
+          def install_dependencies
+            check_node_version!
+
+            output, status = ctx.capture2e("npm install --no-audit --no-optional --legacy-peer-deps --loglevel error")
+            raise Errors::DependencyInstallError, output unless status.success?
+          end
+
+          def dependencies_installed?
+            # Assuming if node_modules folder exist at root of script folder, all deps are installed
+            ctx.dir_exist?("node_modules")
+          end
+
+          def metadata
+            unless @ctx.file_exist?(METADATA_FILE)
+              msg = @ctx.message("script.error.metadata_not_found_cause", METADATA_FILE)
+              raise Domain::Errors::MetadataNotFoundError, msg
+            end
+
+            raw_contents = File.read(METADATA_FILE)
+            Domain::Metadata.create_from_json(@ctx, raw_contents)
+          end
+
+          private
+
+          def check_node_version!
```

Comment:
cc @Thenura you may have to update #1454, or create a new one to update this task runner following that PRs pattern when its approved